flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -328,7 +332,17 @@ private void registerInputMetrics(MetricGroup inputGroup, MetricGroup buffersGro
 			InputGateMetrics.registerQueueLengthMetrics(inputGroup, inputGates);
 		}
 		buffersGroup.gauge(METRIC_INPUT_QUEUE_LENGTH, new InputBuffersGauge(inputGates));
-		buffersGroup.gauge(METRIC_INPUT_POOL_USAGE, new InputBufferPoolUsageGauge(inputGates));
+
+		if (config.isCreditBased()) {
```

Comment:
For exclusive case there are no buffer pool actually. So maybe we could break down into `ExclusiveBuffersUsageGauge`, `FloatingBuffersUsageGauge`, `InputBuffersUsageGauge` instead?